10371.xml

```diff
@@ -0,0 +1,85 @@
+			</Item>
+			<Item ID="1">
+				<Name>Reboot Error Message</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Mandatory</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description><![CDATA[Containing the error message of the last reboot failure. This value MUST be set to an empty string on exec on /3/0/4, ideally before to send the response for Reboot resource (3/0/4) to avoid any race conditions]]></Description>
+			</Item>
+			<Item ID="2">
+				<Name>Reset Factory State</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Integer</Type>
+				<RangeEnumeration>0..3</RangeEnumeration>
+				<Units></Units>
+				<Description><![CDATA[ 		Containing the state of reset factory operation. At start-up the value should be 0 (initial-state). Once the reset factory process is scheduled (exec on /3/O/5), this value must be set to 1 (resetting), it is advised to do that before to send response of Reset Factory resource (/3/0/5) to avoid any race conditions. If reset fails for any reason value must be set to 2 (reset error). If reset succeeds, state should be set to 3 (reset). If the reset factory need a reboot, reboot state(/10371/0/0) resource SHOULD also be updated accordingly. If reset factory need a reboot, in case of successful reset, the value MUST be set to 3 (reset) at device start-up. If it helps implementation, this is tolerated to have this resource set to 3 (reset) at first device start-up.]]></Description>
+			</Item>
+			<Item ID="3">
+				<Name>Reset Factory Error Message</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description><![CDATA[Containing the error message of the last reset factory failure. This value MUST be set to an empty string on exec on /3/0/5, ideally before to send the response for Reset Factory resource (3/0/5) to avoid any race conditions]]></Description>
+			</Item>
+		</Resources>
+		<Description2><![CDATA[This LWM2M object provides features to monitor Reboot(/3/0/4) and Reset Factory(/3/0/5) operation available on LWM2M Device (3) Object, allowing to know if a reboot or a reset factory is a success or a failure and why it failed. This is strongly recommended to use separated response for those resources (/3/0/4 and /3/0/5) and wait for response ACK before to execute/schedule the operation (reboot or reset factory). This prevents LWM2M server to consider the request as a failure in case response is lost. This situation could be annoying as server could decide to resent the request later which could bring to unexpected reboot or reset factory.]]></Description2>
```

Comment:
As others have mentioned, instead of extending, it would be good to directly update /3 with these resources. Maybe DM can have a look and add it to the spec at some point?